static/js/src/public/details/integrations/types.ts

```diff
@@ -3,6 +3,7 @@ export interface IInterfaceData {
   interface: string;
   description: string;
   type?: "provides" | "requires";
+  optional: string;
```

Comment:
Should optional be optional? It might not be in the data at all, right? So this should potentially be `optional?:` also is it a string or is it a boolean? If it's a string, we should know what the values can be so you can do something similar to the `type?: "provides" | "requires"` on the line above.

Reply:
From the data we are getting from the `["store_front"]` key, the `optional` field is implemented as a boolean value. I made changes accordingly and updated the field to `required` as this will be the standard, as Lukas mentioned.